wrench/src/main.rs

```diff
@@ -175,97 +320,138 @@ fn main() {
+                    break 'outer;
+                }
 
-            glutin::Event::KeyboardInput(kstate, _scan_code, maybe_vk) => {
-                if kstate == ElementState::Pressed {
-                    if let Some(vk) = maybe_vk {
-                        match vk {
-                            VirtualKeyCode::Escape | VirtualKeyCode::Q => {
-                                break;
-                            },
-                            VirtualKeyCode::P => {
-                                profiler = !profiler;
-                                wrench.renderer.set_profiler_enabled(profiler);
-                            },
-                            VirtualKeyCode::L => {
-                                do_loop = !do_loop;
-                            },
-                            VirtualKeyCode::Left => {
-                                thing.prev_frame();
-                            },
-                            VirtualKeyCode::Right => {
-                                thing.next_frame();
-                            },
-                            VirtualKeyCode::H => {
-                                show_help = !show_help;
-                            },
-                            _ => ()
+                glutin::Event::KeyboardInput(kstate, _scan_code, maybe_vk) => {
+                    if kstate == ElementState::Pressed {
+                        if let Some(vk) = maybe_vk {
+                            match vk {
+                                VirtualKeyCode::Escape | VirtualKeyCode::Q => {
+                                    break 'outer;
+                                },
+                                VirtualKeyCode::P => {
+                                    profiler = !profiler;
+                                    wrench.renderer.set_profiler_enabled(profiler);
+                                },
+                                VirtualKeyCode::L => {
+                                    do_loop = !do_loop;
+                                },
+                                VirtualKeyCode::Left => {
+                                    thing.prev_frame();
+                                },
+                                VirtualKeyCode::Right => {
+                                    thing.next_frame();
+                                },
+                                VirtualKeyCode::H => {
+                                    show_help = !show_help;
+                                },
+                                _ => ()
+                            }
                         }
                     }
                 }
+                _ => ()
             }
-            _ => ()
         }
     }
+
+    if is_headless {
+        let mut pixels = gl::read_pixels(0, 0,
+                                         size.width as gl::GLsizei,
+                                         size.height as gl::GLsizei,
+                                         gl::RGB, gl::UNSIGNED_BYTE);
```

Comment:
Perhaps - I'm not entirely sure. But it's only done once when writing the screenshot in test mode, so probably not a big deal.